processing/src/main/java/org/apache/druid/query/PrioritizedQueryRunnerCallable.java

```diff
@@ -17,20 +17,15 @@
  * under the License.
  */
 
-package org.apache.druid.guice.annotations;
-
-import com.google.inject.BindingAnnotation;
-
-import java.lang.annotation.ElementType;
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
-import java.lang.annotation.Target;
+package org.apache.druid.query;
 
 /**
+ * An implementation of {@link PrioritizedCallable} that also let's caller get access to associated {@link QueryRunner}
+ * It is used in implementations of {@link QueryRunnerFactory}
+ * @param <T>
+ * @param <V>
  */
-@BindingAnnotation
-@Target({ElementType.FIELD, ElementType.PARAMETER, ElementType.METHOD})
-@Retention(RetentionPolicy.RUNTIME)
-public @interface Processing
+public interface PrioritizedQueryRunnerCallable<T, V> extends PrioritizedCallable<T>
 {
+  QueryRunner<V> getRunner();
```

Comment:
How will this method be used? I only see it used in a test in this PR.

Reply:
This method can be used by the extensions to get the runner that the given query execution task corresponds to. That in turn can be used to fetch any state associated with the QueryRunner such as the segment info for example.

Reply:
sure thing.